Allow changing favicon and logo in back-office

Paphos opened the pull request:
- Favicon and logo are no longer related to the template and can be changed in the back-office in Configuration / Store.
- The file name of the logo and favicon is stored in database in the "config" table with the keys "favicon_file" and "logo_file".
- Default Thelia favicon and logo are stored in local/media/images/store and used if no name is provided in the "config" table.
- Smarty plugins {logo} and {favicon} are used to get the image path.

roadster31 commented on core/lib/Thelia/Form/ConfigStoreForm.php:
```diff
@@ -201,6 +201,37 @@ protected function buildForm()
+                    'required' => false,
+                    'constraints' => [
+                        new Constraints\Image(array(
+                            'mimeTypes' => 'image/png'
```
Allowing ICO and PNG files would be a good idea.

roadster31 commented on local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php:
```diff
@@ -808,6 +928,8 @@ public function getPluginDescriptors()
             new SmartyPluginDescriptor('function', 'module_config', $this, 'moduleConfigDataAccess'),
             new SmartyPluginDescriptor('function', 'coupon', $this, 'couponDataAccess'),
+            new SmartyPluginDescriptor('function', 'favicon', $this, 'faviconDataAccess'),
+            new SmartyPluginDescriptor('function', 'logo', $this, 'logoDataAccess'),
```
Instead of creating two functions `logo` and `favicon`, a `template` function with an argument would be more future-proof :

`{template attr="logo"}` and `{template attr="favicon"}`

Paphos commented:
- ICO files are now allowed.
- Since it's not related to the template, I refactored the `{favicon}` and `{logo}` smarty functions into a `{media type="favicon|logo"}` smarty block

roadster31 commented:
What about `{local_media}`, which is more specific, and clearly means that media are local to this Thelia installation ?

Could you please also update [the documentation](https://github.com/thelia/thelia.github.io), to add information about these two functions (in the Templates / Smarty section) ?

Additionally, the default email template should benefits from these new features, as the default logo and banner is the following

![img-2017-08-25_12-19-48](https://user-images.githubusercontent.com/2197734/29710507-e3dc94e8-8990-11e7-9d7b-df617b342b29.png)

Maybe a` {local_media type="banner"} `?

Paphos commented:
- Changed smarty block `{media}` to `{local_media}`
- New form field allowing to change the banner image in the default e-mail template. Updated smarty block with : `{local_media type="banner"}`
- The field 'Logo' is now changing the logo in the e-mail template as well.
- Versionning the default banner image in local/media/images/store/banner.jpg

I'll update the documentation soon ;)

roadster31 commented:
Thanks :)

zorn-v commented:
There was very bad decision to merge it without testing. Now try to install and use 2.3.4

roadster31 commented:
I did. What is the problem ?

zorn-v commented:
Try to install thelia. Then change version to 2.3.4 (coz there is no another variant). And you got noy working shop coz favicon.png and logo.png "not found"
